### ISSUE TITLE
Петровичу нужен новый хозяин? 

### ISSUE THREAD

**Issue by emejibka** [petrovich/petrovich-net#7]

**Comment by quozd** [comment]
Да, было бы очень круто. У меня к сожалению нет ни времени, ни желания этим заниматься, т.к. я уже совсем отказалсся от .NET.

Отдам репу в добрые руки :)

**Comment by emejibka** [comment]
Отлично, я могу поддержать проект. Есть какие-нибудь требования к хранителю?

**Comment by quozd** [comment]
Кроме желания - никаких :) Дал тебе права на запись :) Спасибо

**Comment by mustiks** [comment]
Проект совсем заброшен?

**Comment by quozd** [comment]
Да, есть желание развивать?

**Comment by mustiks** [comment]
Да, т.к. буду использовать в нескольких проектах.

**Comment by quozd** [comment]
Вот жеж, я вышел из организации и теперь у меня нет прав.
cast @dustalov

**Comment by dustalov** [comment]
@quozd, добавил.

**Comment by quozd** [comment]
@dustalov, добавь, пожалуйста в мемберы @mustiks

**Comment by dustalov** [comment]
@mustiks, добро пожаловать!

**Comment by mustiks** [comment]
@dustalov, Спасибо!

**Comment by mustiks** [comment]
@quozd  А как быть с Nuget? Хотелось бы там тоже обновлять версии.

**Comment by quozd** [comment]
По хорошему надо сделать аккаунт для петровича и запаблишить туда. Сейчас оно на моем личном аккаунте, оттуда надо удалить.

**Comment by mikhail-barg** [comment]
Коллеги, добрый день! Я для производственных нужд сочинил альтернативный порт Петровича на C#: https://github.com/mikhail-barg/petrovich-net-lite
Отличия, которые были важны мне:
* измененный (более простой на мой взгляд) API библиотеки;
* поддержка правил Petrovich'а версии v1 (https://github.com/petrovich/petrovich-ruby/issues/48) включая определение пола через правила;
* нет завязки на внешние библиотеки (вместо чтения из YAML, правила читаются из JSON велосипедным парсером).

Поскольку апи поменялось сильно, то пушить в текущий проект наверное смысла нет. Однако мне мое решение нравится, и хотелось бы его популяризировать ) Как вы смотрите на включение проекта в официальные?

**Comment by dustalov** [comment]
Если @mikhail-barg готов поддерживать порт, то я согласен.

**Comment by mikhail-barg** [comment]
Да, свой порт я, естественно, готов поддерживать.

**Comment by mikhail-barg** [comment]
@dustalov Каким образом лучше это сделать? Форкнуть мой репозиторий в организацию (как например порты на php и c++) или создать новый репозиторий в организации и сделать PR в него?

**Comment by dustalov** [comment]
Отмечу, что здесь очень неудобно следить за упоминаниями. Нам важно сохранить обратную совместимость портов в NuGet: <https://www.nuget.org/packages/NPetrovich/>.

Предлагаю следующий процесс:

* @quozd и @mikhail-barg подтверждают, что согласны на передачу;
* @quozd добавляет права @mikhail-barg в NuGet;
* @mikhail-barg добавляет в свой код API обратной совместимости с пометкой `[Obsolete]`;
* @mikhail-barg отправляет свои коммиты в ветку `1.1` этого репозитория;
* @mikhail-barg проверяет, что всё хорошо, заменяет этой веткой `master` и выпускает новую версию `1.1.0` пакета в NuGet.

**Comment by mikhail-barg** [comment]
@dustalov я честно говоря не хотел бы ломать обратную совместимость, но и поддерживать текущее API NPetrovich мне не нравится. Я поэтому предлагаю завести альтернативный порт NPetrovichLite (или другое имя по выбору) .

Внедрять текущее АПИ в мой порт мне не хочется по тем же причинам, по которым я решил писать отдельный порт:
* Текущее АПИ NPetrovich на мой взгляд вообще не отражает реальные use-case'ы. В моем АПИ весь функционал доступен через два метода, инстанс класса нужен для того, чтобы загрузить правила. В АПИ NPetrovich все выглядит куда более громоздким.
* NPetrovich тянет за собой внешние зависимости для чтения YAML. Моя реализация самодостаточна, что важно для меня (я использую код в CLR-процедурах SQL-сервера, и лишние ассембли мне там ни к чему).

В виду того, что все вышесказанное скорее дело вкуса, я бы не хотел портить уже существующий NPetrovich своими переделками, но и нагружать свою реализацию весьма объемным и ненужным интерфейсом я бы не хотел. К тому же, чтобы сохранить полную обратную совместимость мне придется вводить поддержку YAML, что вроде тоже ни к чему.

Поэтому я и предлагаю завести второй официальный порт на C#. Надеюсь это не выглядит излишне нагло.

**Comment by dustalov** [comment]
@mikhail-barg с другой стороны, я только что заметил, что за два года было всего пятьсот скачиваний. Я не против сломать обратную совместимость, если будет написана инструкция по обновлению.

**Comment by andrew-iv** [comment]
Заинтересовался Петровичем.
Хочу присоединиться к вам.
Есть желание его довести его до соответствия ruby версии.
Создал пулл реквест https://github.com/petrovich/petrovich-net/pull/13 Для доработок по определению пола.
Также есть планы актуализации Js версии.

**Comment by dustalov** [comment]
Насколько понимаю, основная трудность — релизы в NuGet. @quozd, возможно ли предоставить другим участникам права на библиотеку [NPetrovich](https://www.nuget.org/packages/NPetrovich/)?

**Comment by mikhail-barg** [comment]
Коллеги, привет. 
Поскольку с нугетом, обновлениями и прочими изменениями так ничего и не решилось, на сколько я понимаю, я зарелизил свой [лайтвейтный порт](https://github.com/mikhail-barg/petrovich-net-lite) в NuGet: https://www.nuget.org/packages/NPetrovichLite/
Порт актуализирован до (практически) последней версии основного проекта, используются последние правила, ну и плюс на мой взгляд у него есть ряд преимуществ по сравнению с официальным NPetrovichNet (о них я писал чуть выше).

Кстати, я бы хотел в своем порте использовать официальную иконку. Есть ли у кого-то возражения по этому поводу?

**Comment by quozd** [comment]
btw, нугет был передан @andrew-iv пару месяцев назад.

**Comment by andrew-iv** [comment]
В последенем релизе было добавлено увеличение производительности, определение пола по ФИО, обновлен словарь.

**Comment by dustalov** [comment]
@mikhail-barg в своё время Андрей Бильжо разрешил использовать изображение Петровича для нашей библиотеки.

@andrew-iv, @quozd, @mikhail-barg было бы здорово оставить только один вариант библиотеки для избежания фрагментации.

**Comment by mikhail-barg** [comment]
@dustalov 
> в своё время Андрей Бильжо разрешил использовать изображение Петровича для нашей библиотеки.

Да, я читал эти тикеты, но просто поскольку моя версия не официальная, то не совсем понятно, насколько это разрешение к ней относится. Надеюсь, что ни у кого проблем это не вызовет.

> было бы здорово оставить только один вариант библиотеки для избежания фрагментации.

По этому поводу я свои мысли излагал выше в этой теме. Мне текущий интерфейс официального порта не нравится, но непонятно, на каких основаниях я мог бы навязать свое мнение другим разработчикам.

Плюс непонятно, что делать с обратной совместимостью, если брать мой интерфейс. Вероятно, можно было бы зарелизить мажорную версию, которая предполагала бы breaking changes. В таком случае желающие могли бы остаться на старой версии.

**Comment by ujifgc** [comment]
Обычно делают легаси-модуль с трансляцией вызовов, а потом его устаревают.

**Comment by dustalov** [comment]
Считаю, что основным интерфейсом должно стать решение, которое будет активно поддерживаться. Поэтому вопрос обращается к разработчикам параллельных портов.

Насколько знаю, в C# есть поддержка устаревания: `[Obsolete]`.
